lib/msf/core/rpc/v10/rpc_module.rb

```diff
@@ -492,6 +522,14 @@ def rpc_options(mtype, mname)
   #  rpc.call('module.execute', 'exploit', 'multi/handler', opts)
   def rpc_execute(mtype, mname, opts)
     mod = _find_module(mtype,mname)
+    dispatcher = dispatcher(mtype)
+
+    # Choose a default payload when the module is used, not run
+    if mod.datastore['PAYLOAD']
+      opts['PAYLOAD'] = mod.datastore['PAYLOAD']
+    elsif dispatcher.respond_to?(:choose_payload)
+      opts['PAYLOAD'] = dispatcher.choose_payload(mod)
+    end
```

Comment:
I'm not sure if this logic makes sense in any of the module types, other than exploit? 👀

Reply:
Hm, are you thinking that should just be extracted into `_run_exploit` instead?